cmd/tailscale/cli/cli_test.go

```diff
@@ -621,9 +621,16 @@ func TestPrefsFromUpArgs(t *testing.T) {
 		{
 			name: "error_long_hostname",
 			args: upArgsT{
-				hostname: strings.Repeat("a", 300),
+				hostname: strings.Repeat(strings.Repeat("a", 63)+".", 5),
 			},
-			wantErr: `hostname too long: 300 bytes (max 256)`,
+			wantErr: `hostname too long: 320 bytes (max 255)`,
```

Comment:
additional test cases for the boundary may be good to verify no off-by-one errors.